### PR TITLE
🔄 From Aesara: 1362: "Add `HalfNormalRV` JAX implementation"

### PR DESCRIPTION
**Pull request by twiecki**
Downstreaming https://github.com/aesara-devs/aesara/pull/1362. PR port done by downstream_pr.sh script.

### CODE REVIEW

File: pytensor/link/jax/dispatch/random.py [hunk 1]
```diff
@@ -251,6 +251,28 @@ def sample_fn(rng, size, dtype, *parameters):
...
+    def sample_fn(rng, size, dtype, *parameters):
+        rng_key = rng["jax_state"]
+        rng_key, sampling_key = jax.random.split(rng_key, 2)
+        (
+            loc,
+            scale,
+        ) = parameters
+        sample = (
+            loc
+            + jax.random.truncated_normal(sampling_key, 0.0, jax.numpy.inf, size, dtype)
```
Comment (ricardoV94) [inline pytensor/link/jax/dispatch/random.py:267]: This seems inefficient, we can just take the absolute of the normal centered at 0
```suggestion
            + jax.numpy.abs(jax.random.normal(sampling_key, size, dtype))
```

File: pytensor/link/jax/dispatch/random.py [hunk 2]
```diff
@@ -251,6 +251,24 @@ def sample_fn(rng, size, dtype, *parameters):
...
+def jax_sample_fn_halfnormal(op):
+    """JAX implementation of `HalfNormalRV`."""
+
+    def sample_fn(rng, size, dtype, *parameters):
+        rng_key = rng["jax_state"]
+        rng_key, sampling_key = jax.random.split(rng_key, 2)
+        (
+            loc,
+            scale,
+        ) = parameters
```
Comment (twiecki) [inline pytensor/link/jax/dispatch/random.py:264]:
```suggestion
        loc, scale = parameters
```

Comment (twiecki) [reply pytensor/link/jax/dispatch/random.py:264]: Reformat

File: pytensor/link/jax/dispatch/random.py [hunk 3]
```diff
@@ -251,6 +251,21 @@ def sample_fn(rng, size, dtype, *parameters):
...
 
+@jax_sample_fn.register(aer.HalfNormalRV)
+def jax_sample_fn_halfnormal(op):
+    """JAX implementation of `HalfNormalRV`."""
+
+    def sample_fn(rng, size, dtype, *parameters):
+        rng_key = rng["jax_state"]
+        rng_key, sampling_key = jax.random.split(rng_key, 2)
+        loc, scale = parameters
+        sample = loc + jax.numpy.abs(jax.random.normal(sampling_key, size, dtype) * scale
```
Comment (twiecki) [inline pytensor/link/jax/dispatch/random.py:262]:
```suggestion
        sample = loc + jax.numpy.abs(jax.random.normal(sampling_key, size, dtype)) * scale
```

**Comment by ricardoV94** [comment]
Haha, pre-commit and editing via comments sure is a wild ride :D

File: pytensor/link/jax/dispatch/random.py [hunk 4]
```diff
@@ -251,6 +251,21 @@ def sample_fn(rng, size, dtype, *parameters):
...
 
+@jax_sample_fn.register(aer.HalfNormalRV)
+def jax_sample_fn_halfnormal(op):
+    """JAX implementation of `HalfNormalRV`."""
+
+    def sample_fn(rng, size, dtype, *parameters):
+        rng_key = rng["jax_state"]
+        rng_key, sampling_key = jax.random.split(rng_key, 2)
+        loc, scale = parameters
+        sample = loc + jax.numpy.abs(jax.random.normal(sampling_key, size, dtype)) * scale
```
Comment (twiecki) [inline pytensor/link/jax/dispatch/random.py:262]:
```suggestion
        sample = (
            loc + jax.numpy.abs(jax.random.normal(sampling_key, size, dtype)) * scale
        )
```

**Pull request merged**